src/core/public/chrome/ui/header/collapsible_nav_header.tsx

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import React, { useState } from 'react';
+import useObservable from 'react-use/lib/useObservable';
+import {
+  EuiContextMenu,
+  EuiPopover,
+  EuiIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiText,
+  EuiCollapsibleNavGroup,
+} from '@elastic/eui';
+import {
+  HttpStart,
+  WorkspaceStart,
+  WorkspaceAttribute,
+  MANAGEMENT_WORKSPACE,
+} from '../../../../public';
+import { InternalApplicationStart } from '../../../application';
+import { formatUrlWithWorkspaceId } from '../../../utils';
+import {
+  WORKSPACE_CREATE_APP_ID,
+  WORKSPACE_LIST_APP_ID,
+  WORKSPACE_OVERVIEW_APP_ID,
+} from '../../constants';
+
+interface Props {
+  workspaces: WorkspaceStart;
+  basePath: HttpStart['basePath'];
+  getUrlForApp: InternalApplicationStart['getUrlForApp'];
+}
+
+function getFilteredWorkspaceList(
+  workspaces: WorkspaceStart,
+  workspaceList: WorkspaceAttribute[],
+  currentWorkspace: WorkspaceAttribute | null
+): WorkspaceAttribute[] {
+  // list top5 workspaces except management workspace
+  let filteredWorkspaceList = workspaceList
+    .filter((workspace) => workspace.id !== MANAGEMENT_WORKSPACE)
+    .slice(0, 5);
+  if (currentWorkspace) {
+    // current workspace located at the top of workspace list
+    filteredWorkspaceList = filteredWorkspaceList.filter(
+      (workspace) => workspace.id !== currentWorkspace.id
+    );
+    filteredWorkspaceList.unshift(currentWorkspace);
+    if (filteredWorkspaceList.length > 5) {
+      filteredWorkspaceList.pop();
+    }
+  }
+  return filteredWorkspaceList;
+}
+
+export function CollapsibleNavHeader({ workspaces, getUrlForApp, basePath }: Props) {
```

Comment:
@SuZhou-Joe @ruanyl Do we need to move these workspace related code to workspace plugin?

Reply:
@wanglam Any suggestion on how to render it in left menu if moved to plugin folder?

Reply:
Maybe something like this:

```typescript
// src/core/public/workspace/workspaces_service.ts
class WorkspaceService {
  private _renderSelector;

  renderSelector() {
    if (this._renderSelector) {
      this._renderSelector()
    }
  }

  onRenderSelector(render) {
    this._renderSelector = render
  }
}

// src/plugins/workspace/public/plugin.ts
class WorkspacePlugin {
  setup(core) {
    core.workspaces.onRenderSelector(mountSelector)
  }
}

function mountSelector() {
  return <CollapsibleNavHeader />
}
```
Does it look good to you?

Reply:
Looks good to me.

Reply:
Should this change be included in this PR?